packages/sdk/src/evm/contracts/smart-contract.ts

```diff
@@ -148,19 +148,26 @@ export class SmartContract<TContract extends BaseContract = BaseContract>
     return assertEnabled(this.detectErc1155(), FEATURE_EDITION);
   }
 
+  private _chainId: number;
+  get chainId() {
+    return this._chainId;
```

Comment:
Why not just `public chainId: number` directly?

Reply:
because then u can technically set it